scipp-imaging-2D.ipynb

```diff
@@ -0,0 +1,496 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Adds the component info needed for converting units\n",
+    "sc.compat.mantid.load_component_info(stitched, instrument_file)\n",
+    "stitched"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Note: this cell is currently a workaround until \n",
+    "# https://github.com/scipp/scipp/issues/823 is done.\n",
+    "# Then it should be possible to group the positions label directly.\n",
+    "\n",
+    "# makes the position a DataConstProxy otherwise groupby won't take it\n",
+    "stitched[\"position\"] = stitched.labels[\"position\"]\n",
+    "dm1d = sc.groupby(stitched[\"position\"], \"detector_mapping\", Dim.Row)\n",
+    "position = dm1d.mean(Dim.Spectrum)\n",
+    "\n",
+    "# can't do stitched.labels[\"position\"] = position because Labels won't take a DataArray\n",
+    "# also can't do stitched.labels[\"position\"] = sc.Variable(position) because then sc.convert \n",
+    "# thinks stitched.labels[\"position\"] is dimensionless (as it's actually still a DataArray)\n",
+    "\n",
+    "pos = np.zeros((position.shape[0], 3))\n",
+    "for i, val in enumerate(position.values):\n",
+    "    pos[i, :] = val\n",
+    "\n",
+    "# finally add it back!\n",
+    "stitched.labels[\"position\"] = sc.Variable(position.dims, pos, unit=sc.units.m, dtype=position.dtype)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "stitched[\"normalized_wl\"] = sc.neutron.convert(stitched[\"normalized_grpd\"], Dim.Tof, Dim.Wavelength)\n",
+    "stitched"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "sc.plot.plot(stitched[\"normalized_wl\"][Dim.Row, 0]) if do_plots else None"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "x_coords = stitched[\"normalized_wl\"].coords[stitched[\"normalized_wl\"].dims[0]]\n",
+    "x_dim = stitched[\"normalized_wl\"].dims[0]\n",
+    "x = x_coords.values\n",
+    "\n",
+    "# Mantid expects the data in a different shape\n",
+    "# which is spectrum as outer-most dimension.\n",
+    "y = np.transpose(stitched[\"normalized_wl\"].values)\n",
+    "\n",
+    "ws = sc.compat.mantid.to_workspace_2d(x, y, None, x_dim)\n",
+    "ws"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# If you want to save the workspace uncomment this line\n",
+    "SaveNexus(ws, \"scipp_normalized_wl.nxs\")"
```

Comment:
`SaveNexusESS` should probably be used. Note that your `to_workspace_2d` would need to generate an instrument and attach it to output from scipp to make this fully useful.

Reply:
I'll add this as a change for https://github.com/scipp/scipp/pull/830

Reply:
Its unfortunate that we didn't get round to bypassing instrument 1.0 in the instrument 2.0 construction otherwise you could have created the instrument directly from your arrays.